Tests/NIOHTTP2Tests/ConfiguringPipelineAsyncMultiplexerTests.swift

```diff
@@ -0,0 +1,150 @@
+            }
+
+            context.channel.writeAndFlush(responseFramePayload, promise: nil)
+
+            context.fireChannelRead(data)
+        }
+    }
+
+    class SimpleRequest: ChannelInboundHandler, ChannelOutboundHandler {
+        typealias InboundIn = HTTP2Frame.FramePayload
+        typealias OutboundIn = HTTP2Frame.FramePayload
+
+        func writeRequest(context: ChannelHandlerContext) {
+            context.channel.writeAndFlush(requestFramePayload, promise: nil)
+        }
+
+        func channelActive(context: ChannelHandlerContext) {
+            self.writeRequest(context: context)
+            context.fireChannelActive()
+        }
+
+        func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+            let frame = self.unwrapInboundIn(data)
+            switch frame {
+            case .headers:
+                break
+            default:
+                fatalError("unexpected frame type: \(frame)")
+            }
+            context.fireChannelRead(data)
+        }
+    }
+
+    func testBasicPipelineCommunicates() async throws {
+        let requestCount = 100
+
+        let serverRecorder = InboundFramePayloadRecorder()
+        let (_, clientMultiplexer) = try await assertNoThrowWithValue(
+            try await self.clientChannel.configureHTTP2PipelineAsync(
+                mode: .client, connectionConfiguration: .init(), streamConfiguration: .init()) { channel in self.serverChannel.eventLoop.makeSucceededFuture(channel) }.get()
+        )
+
+        let (serverStreamChannels, _) = try await assertNoThrowWithValue(
+            try await self.serverChannel.configureHTTP2PipelineAsync(
+                mode: .server, connectionConfiguration: .init(), streamConfiguration: .init()) { channel in
+                    channel.pipeline.addHandlers([OKResponder(), serverRecorder]).map { _ in channel }
+                }.get()
+        )
+
+        try await assertNoThrow(try await self.assertDoHandshake(client: self.clientChannel, server: self.serverChannel))
+
+
+        try await withThrowingTaskGroup(of: Int.self, returning: Void.self) { group in
+            // server
+            group.addTask {
+                var serverInboundChannelCount = 0
+                for try await _ in serverStreamChannels {
+                    serverInboundChannelCount += 1
+                }
+                return serverInboundChannelCount
+            }
+
+            // client
+            for _ in 0 ..< requestCount {
+                // Let's try sending some requests
+                let streamChannel = try await clientMultiplexer.createStreamChannel() { channel in
+                    return channel.pipeline.addHandlers([SimpleRequest(), InboundFramePayloadRecorder()]).map {
+                        return channel
+                    }
+                }
+
+                let clientRecorder = try await streamChannel.pipeline.handler(type: InboundFramePayloadRecorder.self).get()
```

Comment:
That's also not safe since the handler is not `Sendable`. You should be able to do the same trick here with the `AsyncStream`